### PR TITLE
Goext timeouts

### PR DESCRIPTION
**Pull request by p-kozlowski**
- pass cancelable context to golang handlers
- cancel the context on extension execution timeout
- pass context to sync.Watch

### CODE REVIEW

File: extension/goplugin/environment_test.go [hunk 1]
```diff
@@ -272,5 +275,65 @@ var _ = Describe("Environment", func() {
...
+				done <- true
+			}()
+
+			closeNotifier.Close()
+
+			Eventually(done).Should(Receive())
+		})
+
+		It("should exit gracefully on global execution timeout", func() {
+			env.LoadExtensionsForPath(manager.Extensions, time.Millisecond*100, nil, "wait_for_context_cancel")
```
Comment (przemyslaw-dobrowolski-cl) [inline extension/goplugin/environment_test.go]: Expect here?

File: extension/goplugin/environment_test.go [hunk 2]
```diff
@@ -272,5 +275,65 @@ var _ = Describe("Environment", func() {
...
+
+			Eventually(done, time.Millisecond*500).Should(Receive())
+		})
+
+		It("should exit gracefully on path execution timeout", func() {
+			timeLimits := []*schema.PathEventTimeLimit{
+				schema.NewPathEventTimeLimit(".*", "wait_for_context_cancel", time.Millisecond*100),
+			}
+
+			env.LoadExtensionsForPath(manager.Extensions, 0, timeLimits, "wait_for_context_cancel")
```
Comment (przemyslaw-dobrowolski-cl) [inline extension/goplugin/environment_test.go]: Expect here?

**Review by przemyslaw-dobrowolski-cl — APPROVED** [review]
LGTM

File: extension/goplugin/environment_test.go (hunk 1 above)
Comment (p-kozlowski) [reply extension/goplugin/environment_test.go]: done

File: extension/goplugin/environment_test.go (hunk 2 above)
Comment (p-kozlowski) [reply extension/goplugin/environment_test.go]: done

**Comment by p-kozlowski** [comment]
moved to https://github.com/cloudwan/gohan/pull/530

**Pull request closed without merging**